main/ui/src/main/java/org/cryptomator/ui/preferences/GeneralPreferencesController.java

```diff
@@ -198,10 +200,10 @@ public KeychainBackend fromString(String string) {
 
 	private static class ToggleAutoStartTask extends Task<Void> {
 
-		private final AutoStartStrategy autoStart;
+		private final AutoStartProvider autoStart;
 		private final boolean enable;
 
-		public ToggleAutoStartTask(AutoStartStrategy autoStart, boolean enable) {
+		public ToggleAutoStartTask(AutoStartProvider autoStart, boolean enable) {
```

Comment:
Is it still necessary to enable/disable in an async task? Is the new implementation still potentially long-running or always nearly instant?

Reply:
I would say nearly instantly. On windows you either create or remove a small file in the home directory, this should be possible within milliseconds.

Reply:
What is the situation on mac?

Reply:
Same, just a system call. I don't expect this to take longer than a few ms on any sane system. Windows used to be a little slow due to our former approach that spawned subprocesses to access the registry.

Reply:
Well sine, we are either using jdk's `Files` class or also native code without any subprocesses, i removed the `ToggelAutoStartTask` code.

Resolved in 4402c19